### ISSUE TITLE
[BUG] Moonshine v2 models crash with >= 10 seconds audio input

### ISSUE THREAD

**Issue by zugaldia** [k2-fsa/sherpa-onnx#3471]
### Describe the issue

I downloaded the 2026 Moonshine v2 models from https://github.com/k2-fsa/sherpa-onnx/releases/tag/asr-models (e.g. `sherpa-onnx-moonshine-tiny-en-quantized-2026-02-27.tar.bz2`).

Then I followed the example here to run them on a Java 25 application with the latest Sherpa ONNX version: https://github.com/k2-fsa/sherpa-onnx/blob/master/java-api-examples/NonStreamingDecodeFileMoonshineV2.java.

### Steps to reproduce

I trimmed an audio file to 9 and 10 seconds, respectively, with: `ffmpeg -i jfk.wav -t <seconds> jfk_seconds.wav)`.

I can decode the 9 seconds file, but the 10 seconds file gives the following error:

```
2026-04-05 06:12:25.204258899 [E:onnxruntime:, sequential_executor.cc:572 ExecuteKernel] Non-zero status code returned while running Add node. Name:'/model/decoder/layers.0/encoder_attn/Add' Status Message: /shared/onnxruntime/core/providers/cpu/math/element_wise_ops.h:540 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 2 by 415

2026-04-05 06:12:25.204306218 [E:onnxruntime:, sequential_executor.cc:572 ExecuteKernel] Non-zero status code returned while running If node. Name:'optimum::if' Status Message: Non-zero status code returned while running Add node. Name:'/model/decoder/layers.0/encoder_attn/Add' Status Message: /shared/onnxruntime/core/providers/cpu/math/element_wise_ops.h:540 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 2 by 415

/home/runner/work/sherpa-onnx/sherpa-onnx/sherpa-onnx/csrc/offline-recognizer-moonshine-v2-impl.h:DecodeStream:105 

Caught exception:

Non-zero status code returned while running If node. Name:'optimum::if' Status Message: Non-zero status code returned while running Add node. Name:'/model/decoder/layers.0/encoder_attn/Add' Status Message: /shared/onnxruntime/core/providers/cpu/math/element_wise_ops.h:540 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 2 by 415


Return an empty result. Number of audio samples: 160000
```

### Expected behavior

I expected being able to transcribe longer audio files of at least 30 seconds as you can do with Whisper models.

It's unclear to me if this is a limitation of the original Moonshine models or a limitation of the ORT models bundled with Sherpa. 

### Environment
- `sherpa-onnx` version: v1.12.35
- OS: Ubuntu 24.04 LTS
- Java version: 25
- GPU (if applicable): N/A
- NPU (if applicable): N/A

**Comment by csukuangfj** [comment]
Please.test first.with the moonshine official repo.

You cannot take it for granted that it supports 30s as whisper.

**Comment by csukuangfj** [comment]
@zugaldia 

Since the Moonshine team has not open-sourced the scripts used to export their PyTorch models to ONNX, we are effectively dealing with a "black box" regarding the model's internal graph logic.

The error originates entirely from within the pre-compiled model's architecture, specifically a hardcoded internal broadcast operation, rather than our implementation or the Java API. Because the export logic is proprietary/closed, we cannot modify the graph to fix the dynamic axis scaling from our side.

You can use a VAD to segment your audio. An example can be found at https://github.com/k2-fsa/sherpa-onnx/blob/master/java-api-examples/VadNonStreamingSenseVoice.java
It should be straightforward to replace the sense voice model with a moonshine model.

**Comment by csukuangfj** [comment]
By the way, you are not the only one having errors with moonshine models. 

See also 
- https://github.com/huggingface/optimum/issues/2403

Please ask the moonshine team to fix their models.

**Comment by zugaldia** [comment]
Thank you for the additional information. The Hugging Face ticket seems to indicate this is not an issue with this project. Indeed. I'll follow along upstream.

**Comment by paoloantinori** [comment]
Update for anyone landing here: I re-diagnosed this in #3975. The shipped graphs are not hardcoded-broken; they follow the transformers mask contract and fail only when `encoder_attention_mask` arrives at frame length instead of raw audio length, which is what sherpa's `ForwardDecoder` fed. One-line fix in #3976, verified by decoding a 20.4s clip that returns empty on 1.13.8. Full analysis and reproducer: https://github.com/RisorseArtificiali/anti-vocale/blob/main/docs/research/2026-09-22-moonshine-v2-384-frame-limit.md
